Source/Mosa.FileSystem/IPartitionDevice.cs

```diff
@@ -1,6 +1,6 @@
 // Copyright (c) MOSA Project. Licensed under the New BSD License.
 
-namespace Mosa.DeviceSystem
+namespace Mosa.FileSystem
```

Comment:
But then if I don't I can't access it (can't access DeviceSystem via FileSystem.)

Reply:
The filesystem is dependent on the device system.

Reply:
Not anymore with my PR, since I had to make DeviceSystem dependant on FileSystem to make the new HAL function, and thus remove the DeviceSystem dependency on FileSystem (or else it would've made a cyclic dependency).

Reply:
Well, let's not do that. 

FileSystem consumes resources from the Device systems and not visa versa.

Reply:
How could we do it then?

Reply:
Undo those changes. It should work the the original way.

Reply:
The cause is adding HAL.GetCurrentFileSystem() method at the wrong abstraction level. It messes up the proper dependency order.

Reply:
No no, I mean how can we actually add that function (GetCurrentFileSystem()) if we can't add it in the DeviceSystem? Where can we add it?